### ISSUE TITLE
travis-5.x.cfg is broken

### ISSUE THREAD

**Issue by gyst** [collective/buildout.plonetest#10]
The build downloads 5.0a2 but tries to extract 5.0a3; this fails.

Downloading https://launchpad.net/plone/5.0/5.0a2/+download/Plone-5.0a2-UnifiedInstaller.tgz
download: Extracting package to /home/travis/build/cosent/plonesocial.activitystream/parts/download
Installing install.
tar (child): /home/travis/build/cosent/plonesocial.activitystream/parts/download/Plone-5.0a3-UnifiedInstaller/packages/buildout-cache.tar.bz2: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now

Full traceback see e.g. https://travis-ci.org/cosent/plonesocial.activitystream/builds/53361711
This is a package I just switched from 4.x to 5.x.

**Comment by datakurre** [comment]
Of course, there's also that since Travis-CI started supporting build caches for public repos (given that the build could work without sudo permissions), it's no longer necessary to use a special buildout with UnifiedInstalled based speedup.

I'd consider the current best practice with Travis to just use a normal buildout and

```
    sudo: false
    cache:
      directories:
      - eggs
```

in the Travis yaml.

**Comment by do3cc** [comment]
I agree with @datakurre but added a fix now

**Comment by idgserpro** [comment]
Still have problem. Now it creates the wrong download url:

```bash

Installing download.
Downloading https://launchpad.net/plone/5.1/5.2.0/+download/Plone-5.2.0-UnifiedInstaller.tgz
While:
  Installing download.
Error: Error downloading extends for URL https://launchpad.net/plone/5.1/5.2.0/+download/Plone-5.2.0-UnifiedInstaller.tgz: HTTP Error 404: Not Found
```

[plone-series](https://github.com/collective/buildout.plonetest/blob/master/travis-5.x.cfg#L4) is wrong

**Comment by mauritsvanrees** [comment]
You can change the `plone-series` variable in `travis-5.x.cfg` to 5.2, just like in [travis-5.2.x.cfg](https://github.com/collective/buildout.plonetest/blob/master/travis-5.2.x.cfg).

**Comment by idgserpro** [comment]
> You can change the plone-series variable in travis-5.x.cfg to 5.2, just like in travis-5.2.x.cfg.

Yes, but it would be good to fix travis-5.x.cfg.

**Comment by mauritsvanrees** [comment]
I meant: you can create a PR. :-)

**Comment by idgserpro** [comment]
@mauritsvanrees , I agree with @datakurre  that `travis.cfg` should no longer be used, since we have the `Travis` cache. `bobtemplates.plone` [no longer uses it](https://github.com/plone/bobtemplates.plone/blob/master/bobtemplates/plone/addon/.travis.yml.bob#L26). Perhaps it's best to remove travis.cfg. This'll break some builds but in the end it will be good. Do you agree?

**Comment by mauritsvanrees** [comment]
I wouldn't remove those travis files. Likely many existing projects use it. Removing the files forces them to modernise their test setup, which could be good, but is also irritating. It is a bit like removing packages from PyPI that you think no one in their right mind should use anymore, but that happen to work just fine anyway...

It would be good to first clean up the documentation in `README.rst` to show the current best practice and warn against the use of the `travis*cfg` files.

**Comment by idgserpro** [comment]
The problem is that the file is already broken.

My 0.02c:

- Improve documentation;
- Empty the travis-5.x.cfg travis-5.2.cfg files adding only a comment in the file saying the file is deprecated (so when people have the error on their command line they will try to access the travis cfg file on the rbwoser);
- Put a `DEPRECATED` warning in all older travis files.

We suggest doing it in 5.2 as well since it's a new release and it's expected for things to break.

**Comment by mauritsvanrees** [comment]
Well, `travis-5.2.cfg` is working just fine (as far as I know). So I would not like to deliberately break that.
The immediate problem is that `test-5.x.cfg` uses the https://dist.plone.org/release/5-latest/, which is since very recently pointing to 5.2, but `travis-5.x.cfg` is still on `plone-series` 5.1. That is easily fixable in the latter file. Let me make a PR for that.
Then documentation can be improved, and we could indeed add a DEPRECATED warning in all Travis files.

**Comment by mauritsvanrees** [comment]
Once PR #50 is merged, the immediate problem is fixed, so this issue can be closed.
I have opened a new issue #51 for improving the documentation.

**Comment by mauritsvanrees** [comment]
I have merged PR #52:

- The documentation is fixed. Summary: "Do not use the `travis*.cfg` files."
- `travis-5.x.cfg`has not worked for a while because it half uses 5.1 and half 5.2. I have emptied it, except for a deprecation message.
- `travis-5.2.x.cfg` has never worked, and has been removed.

I close this issue.
